### PR TITLE
Fix: programmatically determines the kms region and guid from the crn

### PR DESCRIPTION
**Pull request by Soaib024**
### Description

https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/issues/189

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

added existing_kms_instance_crn and removed existing_kms_instance_guid and kms_region variable

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by Soaib024** [comment]
Upgrade test failed due to addition of new variable`existing_kms_instance_crn` hence skipping upgrade test
Screenshot 2024-06-07 at 7 36 52 PM

File: solutions/secure/variables.tf [hunk 1]
```diff
@@ -120,10 +114,9 @@ variable "existing_kms_key_crn" {
   default     = null
 }
 
-variable "existing_kms_instance_guid" {
-  description = "The GUID of an existing Hyper Protect or Key Protect instance in the same account as the Elasticsearch database instance. Always used to create an authorization policy and if 'existing_kms_key_crn' is not specified also used to create a KMS root key"
+variable "existing_kms_instance_crn" {
+  description = "The CRN of an existing Hyper Protect or Key Protect instance in the same account as the Elasticsearch database instance. Always used to create an authorization policy and if 'existing_kms_key_crn' is not specified also used to create a KMS root key"
   type        = string
-  default     = null
```
Comment (ocofaigh) [inline solutions/secure/variables.tf:126]: default this back to `null` because its not always required (for example if user passes existing key)

**Review by shemau — APPROVED** [review]
LGTM

File: solutions/secure/variables.tf (hunk 1 above)
Comment (shemau) [reply solutions/secure/variables.tf:126]: Is that true? It is passed to the elastic search instance. It would depend if that is happy to take null...

Comment (shemau) [reply solutions/secure/variables.tf:126]: Where it is used for the authorization policy depending on the value of skip_auth_policy which we allow the consumer to set.

Comment (shemau) [reply solutions/secure/variables.tf:126]: Ah... so it could be null if skipping auth policy and using an existing key. I see. Sorry to be slow.

Comment (Soaib024) [reply solutions/secure/variables.tf:126]: addressed

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**